### PR TITLE
add : authentication pages with functionality

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
### Files to check:
- Login.jsx
- Signup.jsx
- Profile.jsx
- auth-context

Note: I will improve ForgotPassword and Profile pages later

### Live link: https://dazzle-cerashop-jfws9izfe-apurvasawant11.vercel.app/

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 # misc
 .DS_Store
-.env
+# .env
```
Comment (jalaj-gl) [inline .gitignore:16]: Why did you remove env from here? You should keep it.

Comment (jalaj-gl) [reply .gitignore:16]: and and the required env vars in vercel itself

File: src/pages/auth/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  const navigate = useNavigate();
+
+  const signUpHandler = () => {
+    const { email, password, confirmPwd, firstName, lastName } = formData;
+    if (
+      email &&
+      password &&
+      confirmPwd &&
+      firstName &&
+      lastName !== "" &&
```
Comment (jalaj-gl) [inline src/pages/auth/Signup.jsx:26]: why add this check on lastName only? are you positive that other values will be filled

**Review by jalaj-gl — APPROVED** [review]
~.~

File: .gitignore (hunk 1 above)
Comment (jalajcodes) [inline .gitignore]: This should be present in gitignore, any reason for commenting it out?

you can just add the vars in vercel itself

File: src/pages/auth/Signup.jsx (hunk 2 above)
Comment (jalajcodes) [inline src/pages/auth/Signup.jsx]: why this check for lastName only? are we positive that other values will not be blank?

**Review by jalajcodes — APPROVED** [review]
Please ignore my previous comment (from `jalaj-gl`).

Some comments, looks good otherwise.

File: src/pages/auth/Signup.jsx (hunk 2 above)
Comment (ApurvaSawant11) [reply src/pages/auth/Signup.jsx]: Forgot to put parenthesis around them. I have updated it. Thanks for pointing it out!

File: .gitignore (hunk 1 above)
Comment (ApurvaSawant11) [reply .gitignore]: I have added JWT secret in `.env` file and since `env` is added in gitignore, the authentication functionality was not working. Please suggest if there is any alternative

File: src/pages/auth/Signup.jsx [hunk 3]
```diff
@@ -19,11 +19,7 @@ export function Signup() {
   const signUpHandler = () => {
     const { email, password, confirmPwd, firstName, lastName } = formData;
     if (
-      email &&
-      password &&
-      confirmPwd &&
-      firstName &&
-      lastName !== "" &&
+      (email && password && confirmPwd && firstName && lastName) !== " " &&
```
Comment (jalajcodes) [inline src/pages/auth/Signup.jsx:22]: have you tested it? This check will fail.

Comment (jalajcodes) [inline src/pages/auth/Signup.jsx]: I hope you have tested if its working.

File: src/pages/profile/Profile.jsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                MIDC, Mira Road East, Mira Bhayandar, Maharashtra, 401107
+                <span className="display-block pt-0p5 pb-1">
+                  Phone: 1800 0000 000
+                </span>
+                <label htmlFor="default-address2">
+                  <input
+                    id="default-address2"
+                    className="checkbox-field"
+                    type="checkbox"
+                  />{" "}
```
Comment (vrinda-mahajan) [inline src/pages/profile/Profile.jsx:165]: remove these extra curly braces.

File: .gitignore (hunk 1 above)
Comment (jalajcodes) [reply .gitignore]: pinged you in discord

**Review by vrinda-mahajan — APPROVED** [review]
Authentication functionality is working properly. Code is also highly readable. Good to merge.👍

File: src/pages/profile/Profile.jsx (hunk 4 above)
Comment (ApurvaSawant11) [reply src/pages/profile/Profile.jsx:165]: It's there to add space between the checkbox and label
